src/default-request-executor.js

```diff
@@ -33,6 +33,8 @@ class DefaultRequestExecutor extends RequestExecutor {
 
   buildRetryRequest(request, response) {
     const newRequest = deepCopy(request);
+    const delta = this.requestTimeout - (new Date() - request.startTime);
+    newRequest.timeout = delta > 0 ? delta : 0;
```

Comment:
I *think* this is okay because we're just coming from the time checks in `parseResponse`, but it feels like it'll be easy to make mistakes in the future - if delta < 0 (which means that we've exceeded time), we'll set it to 0 (which means that there is no timeout). We should probably wrap this in the same checks we've done elsewhere, i.e. `request.startTime > 0`, etc.

Btw if we move buildRetryRequest to after the timeout, this will be more likely to happen.

Reply:
Need a sanity check, but looking at this again, I don't think we need the `delta > 0` check, because in `parseResponse` we error if `elapsedMs > this.requestTimeout`, which would prevent us from getting into a `delta < 1` state?

Reply:
What about the requestTimeout = 0 case? With the updated logic:
- for requestTimeout = 0, we'll have a negative value
- looking at [node-fetch](https://github.com/bitinn/node-fetch/blob/4068bed356396518e7e9c5cc2b8ccd4c137c2749/src/index.js#L55-L62), if it's a non-zero value it'll just use it for setTimeout. Negative value of setTimeout means it'll reject immediately-ish.